### ISSUE TITLE
pip install fails 2021.3.10  ‘_NSIG’ undeclared

### ISSUE THREAD

**Issue by Julien00859** [YoSTEALTH/Liburing#18]
Hello there, I've been trying to install this library from pypi without success.

TL;DR

      gcc -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -I./libs/liburing/src/include -I/home/julien/.pyenv/versions/3.10.6/envs/onioncorn/include -I/home/julien/.pyenv/versions/3.10.6/include/python3.10 -c ./libs/liburing/src/syscall.c -o build/temp.linux-x86_64-cpython-310/./libs/liburing/src/syscall.o
      ./libs/liburing/src/syscall.c: In function ‘__sys_io_uring_enter’:
      ./libs/liburing/src/syscall.c:62:41: error: ‘_NSIG’ undeclared (first use in this function)
         62 |                                         _NSIG / 8);
            |                                         ^~~~~
      ./libs/liburing/src/syscall.c:62:41: note: each undeclared identifier is reported only once for each function it appears in
      ./libs/liburing/src/syscall.c:63:1: warning: control reaches end of non-void function [-Wreturn-type]
         63 | }
            | ^
      error: command '/usr/bin/gcc' failed with exit code 1

Kernel

    $ uname -r
    5.15.0-67-generic

System dependencies (build-essentials are installed too), I don't think they are necessary as you bundle liburing (the C lib) inside of the source install am I right? 

    $ apt search liburing
    i   liburing-dev - Linux kernel io_uring access library - development files                                         
    i A liburing2    - Linux kernel io_uring access library - shared library      

Python/Pip version

    $ pip -V
    pip 23.0.1 from ... (python 3.10)

Full installation message

    $ pip install liburing
    Collecting liburing
      Using cached liburing-2021.3.10.tar.gz (35 kB)
      Installing build dependencies ... done
      Getting requirements to build wheel ... done
      Preparing metadata (pyproject.toml) ... done
    Collecting cffi
      Using cached cffi-1.15.1-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (441 kB)
    Collecting pycparser
      Using cached pycparser-2.21-py2.py3-none-any.whl (118 kB)
    Building wheels for collected packages: liburing
      Building wheel for liburing (pyproject.toml) ... error
      error: subprocess-exited-with-error
      
      × Building wheel for liburing (pyproject.toml) did not run successfully.
      │ exit code: 1
      ╰─> [78 lines of output]
          <string>:11: DeprecationWarning: distutils Version classes are deprecated. Use packaging.version instead.
          prefix                        /usr
          includedir                    /usr/include
          libdir                        /usr/lib
          libdevdir                     /usr/lib
          relativelibdir
          mandir                        /usr/man
          datadir                       /usr/share
          __kernel_rwf_t                yes
          __kernel_timespec             yes
          open_how                      no
          statx                         yes
          C++                           yes
          has_ucontext                  yes
          CC                            gcc
          CXX                           g++
          running bdist_wheel
          running build
          running build_py
          creating build
          creating build/lib.linux-x86_64-cpython-310
          creating build/lib.linux-x86_64-cpython-310/liburing
          copying liburing/interface.py -> build/lib.linux-x86_64-cpython-310/liburing
          copying liburing/helper.py -> build/lib.linux-x86_64-cpython-310/liburing
          copying liburing/wrapper.py -> build/lib.linux-x86_64-cpython-310/liburing
          copying liburing/__init__.py -> build/lib.linux-x86_64-cpython-310/liburing
          running egg_info
          writing liburing.egg-info/PKG-INFO
          writing dependency_links to liburing.egg-info/dependency_links.txt
          writing requirements to liburing.egg-info/requires.txt
          writing top-level names to liburing.egg-info/top_level.txt
          reading manifest file 'liburing.egg-info/SOURCES.txt'
          reading manifest template 'MANIFEST.in'
          warning: no previously-included files found matching 'libs/liburing/src/include/liburing/compat.h'
          adding license file 'LICENSE.txt'
          writing manifest file 'liburing.egg-info/SOURCES.txt'
          running build_ext
          generating cffi module 'build/temp.linux-x86_64-cpython-310/liburing._liburing.c'
          creating build/temp.linux-x86_64-cpython-310
          building 'liburing._liburing' extension
          creating build/temp.linux-x86_64-cpython-310/libs
          creating build/temp.linux-x86_64-cpython-310/libs/liburing
          creating build/temp.linux-x86_64-cpython-310/libs/liburing/src
          creating build/temp.linux-x86_64-cpython-310/build
          creating build/temp.linux-x86_64-cpython-310/build/temp.linux-x86_64-cpython-310
          gcc -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -I./libs/liburing/src/include -I/home/julien/.pyenv/versions/3.10.6/envs/onioncorn/include -I/home/julien/.pyenv/versions/3.10.6/include/python3.10 -c ./libs/liburing/src/queue.c -o build/temp.linux-x86_64-cpython-310/./libs/liburing/src/queue.o
          ./libs/liburing/src/queue.c: In function ‘_io_uring_get_cqe’:
          ./libs/liburing/src/queue.c:122:23: warning: implicit declaration of function ‘__sys_io_uring_enter2’; did you mean ‘SYS_io_uring_enter’? [-Wimplicit-function-declaration]
            122 |                 ret = __sys_io_uring_enter2(ring->ring_fd, data->submit,
                |                       ^~~~~~~~~~~~~~~~~~~~~
                |                       SYS_io_uring_enter
          ./libs/liburing/src/queue.c: In function ‘io_uring_peek_batch_cqe’:
          ./libs/liburing/src/queue.c:183:17: warning: implicit declaration of function ‘__sys_io_uring_enter’; did you mean ‘SYS_io_uring_enter’? [-Wimplicit-function-declaration]
            183 |                 __sys_io_uring_enter(ring->ring_fd, 0, 0,
                |                 ^~~~~~~~~~~~~~~~~~~~
                |                 SYS_io_uring_enter
          gcc -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -I./libs/liburing/src/include -I/home/julien/.pyenv/versions/3.10.6/envs/onioncorn/include -I/home/julien/.pyenv/versions/3.10.6/include/python3.10 -c ./libs/liburing/src/register.c -o build/temp.linux-x86_64-cpython-310/./libs/liburing/src/register.o
          ./libs/liburing/src/register.c: In function ‘io_uring_register_buffers’:
          ./libs/liburing/src/register.c:22:15: warning: implicit declaration of function ‘__sys_io_uring_register’; did you mean ‘SYS_io_uring_register’? [-Wimplicit-function-declaration]
             22 |         ret = __sys_io_uring_register(ring->ring_fd, IORING_REGISTER_BUFFERS,
                |               ^~~~~~~~~~~~~~~~~~~~~~~
                |               SYS_io_uring_register
          gcc -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -I./libs/liburing/src/include -I/home/julien/.pyenv/versions/3.10.6/envs/onioncorn/include -I/home/julien/.pyenv/versions/3.10.6/include/python3.10 -c ./libs/liburing/src/setup.c -o build/temp.linux-x86_64-cpython-310/./libs/liburing/src/setup.o
          ./libs/liburing/src/setup.c: In function ‘io_uring_queue_init_params’:
          ./libs/liburing/src/setup.c:143:14: warning: implicit declaration of function ‘__sys_io_uring_setup’; did you mean ‘SYS_io_uring_setup’? [-Wimplicit-function-declaration]
            143 |         fd = __sys_io_uring_setup(entries, p);
                |              ^~~~~~~~~~~~~~~~~~~~
                |              SYS_io_uring_setup
          gcc -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -I./libs/liburing/src/include -I/home/julien/.pyenv/versions/3.10.6/envs/onioncorn/include -I/home/julien/.pyenv/versions/3.10.6/include/python3.10 -c ./libs/liburing/src/syscall.c -o build/temp.linux-x86_64-cpython-310/./libs/liburing/src/syscall.o
          ./libs/liburing/src/syscall.c: In function ‘__sys_io_uring_enter’:
          ./libs/liburing/src/syscall.c:62:41: error: ‘_NSIG’ undeclared (first use in this function)
             62 |                                         _NSIG / 8);
                |                                         ^~~~~
          ./libs/liburing/src/syscall.c:62:41: note: each undeclared identifier is reported only once for each function it appears in
          ./libs/liburing/src/syscall.c:63:1: warning: control reaches end of non-void function [-Wreturn-type]
             63 | }
                | ^
          error: command '/usr/bin/gcc' failed with exit code 1
          [end of output]
      
      note: This error originates from a subprocess, and is likely not a problem with pip.
      ERROR: Failed building wheel for liburing
    Failed to build liburing
    ERROR: Could not build wheels for liburing, which is required to install pyproject.toml-based projects

**Comment by Julien00859** [comment]
I've solved the problem installing liburing from git

```
pip install git+https://github.com/YoSTEALTH/Liburing```
